pkg/bgpv1/manager/reconcile.go

```diff
@@ -237,35 +237,43 @@ func neighborReconciler(
 	nset := map[string]*member{}
 
 	// populate set from universe of new neighbors
-	for i, n := range newNeigh {
+	for i := range newNeigh {
+		n := newNeigh[i]
+		// Set neighbor default port for peering if unspecified. This is done by kube-apiserver
+		// but added here for non-k8s environments, tests, etc.
+		n.SetDefaultPeerPort()
```

Comment:
We shouldn't set a default value here. If we need to set default value at runtime, we should consolidate that to https://github.com/cilium/cilium/blob/main/pkg/bgpv1/agent/controller.go#L430.